app/assets/javascripts/queue/queueController.js

```diff
@@ -72,31 +32,23 @@ angular.module('ocWebGui.queue', ['ocWebGui.queue.service', 'ui.router', 'ocWebG
       'values': []
     }];
 
-    function getMaxValPlusOne(i) {
-      var maxVal = d3.max(vm.data[i].values, function (x) { return x.calls; });
-      if (maxVal == null) {
-        return 1;
-      }
-      return maxVal + 1;
-    }
-
     function fetchContactStats() {
+      Settings.getOthers().then(function (others) {
+        vm.otherSettings = others;
+      });
+
       $http.get('contacts/stats.json').then(function (response) {
         var data = response.data;
-        var callsValues = data.calls_by_hour
-          .map(function (calls, hour) { return { hour: hour, calls: calls }; })
-          .filter(function (item) { return item.hour >= 8 && item.hour <= 18; });
 
-        var queueValues = data.average_queue_duration_by_hour
-          .map(function (calls, hour) { return { hour: hour, calls: calls }; })
-          .filter(function (item) { return item.hour >= 8 && item.hour <= 18; });
+        var callsValues = ChartJuttu.mapAndFilter(data.calls_by_hour, vm.otherSettings);
+        var queueValues = ChartJuttu.mapAndFilter(data.average_queue_duration_by_hour, vm.otherSettings);
```

Comment:
It's possible that `vm.otherSettings` is not defined here yet because it might be fetched from the API. To fix this, `contacts/stats.json` could be fetched only after we have the settings. Alternatively both could be fetched at the same using [`$q.all`](https://docs.angularjs.org/api/ng/service/$q#all).